<a href="https://colab.research.google.com/github/priyansh1997/AlmaBetter/blob/main/Priyansh_Tripathi_EDA_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 1.1 Importing Libraries
These libraries will be used for Exploratory Data Analysis.

In [1]:
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px

In [2]:
# Removing warnings
import warnings
warnings.filterwarnings("ignore")

# 1.2 Data Loading
Loading the dataset from google drive.

In [3]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [4]:
play_store_df=pd.read_csv('/content/drive/MyDrive/datasets/Play Store Data.csv')
user_reviews_df=pd.read_csv('/content/drive/MyDrive/datasets/User Reviews.csv')

# 1.3 Data Inspection
Inspection of the data contains information about the data, head/tail part, description of dataset, shape etc.

In [5]:
play_store_df.head()

App  ...   Android Ver
0     Photo Editor & Candy Camera & Grid & ScrapBook  ...  4.0.3 and up
1                                Coloring book moana  ...  4.0.3 and up
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ...  4.0.3 and up
3                              Sketch - Draw & Paint  ...    4.2 and up
4              Pixel Draw - Number Art Coloring Book  ...    4.4 and up

[5 rows x 13 columns]

In [6]:
play_store_df.tail()

App  ...         Android Ver
10836                               Sya9a Maroc - FR  ...          4.1 and up
10837               Fr. Mike Schmitz Audio Teachings  ...          4.1 and up
10838                         Parkinson Exercices FR  ...          2.2 and up
10839                  The SCP Foundation DB fr nn5n  ...  Varies with device
10840  iHoroscope - 2018 Daily Horoscope & Astrology  ...  Varies with device

[5 rows x 13 columns]

In [7]:
user_reviews_df.head()

App  ... Sentiment_Subjectivity
0  10 Best Foods for You  ...               0.533333
1  10 Best Foods for You  ...               0.288462
2  10 Best Foods for You  ...                    NaN
3  10 Best Foods for You  ...               0.875000
4  10 Best Foods for You  ...               0.300000

[5 rows x 5 columns]

In [8]:
user_reviews_df.tail()

App  ... Sentiment_Subjectivity
64290  Houzz Interior Design Ideas  ...                    NaN
64291  Houzz Interior Design Ideas  ...                    NaN
64292  Houzz Interior Design Ideas  ...                    NaN
64293  Houzz Interior Design Ideas  ...                    NaN
64294  Houzz Interior Design Ideas  ...                    NaN

[5 rows x 5 columns]

In [9]:
play_store_df.shape

(10841, 13)

In [10]:
user_reviews_df.shape

(64295, 5)

In [11]:
play_store_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


Here we are having table showing 13 feature names and the count of non-values present in them. Now we will check each column in data cleaning part. 

In [12]:
user_reviews_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 64295 entries, 0 to 64294
Data columns (total 5 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   App                     64295 non-null  object 
 1   Translated_Review       37427 non-null  object 
 2   Sentiment               37432 non-null  object 
 3   Sentiment_Polarity      37432 non-null  float64
 4   Sentiment_Subjectivity  37432 non-null  float64
dtypes: float64(2), object(3)
memory usage: 2.5+ MB


Here also in user reviews dataframe we have 5 features with there counts of non-null values.

In [13]:
play_store_df.columns

Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver'],
      dtype='object')

In [14]:
# Checking the column names and removing the spaces so column will be easier to access 
play_store_df.rename(columns={"Content Rating":"Content_Rating", "Last Updated":"Last_Updated", "Current Ver":"Current_Ver", "Android Ver":"Android_Ver"}, inplace=True)

In [15]:
play_store_df.columns

Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content_Rating', 'Genres', 'Last_Updated', 'Current_Ver',
       'Android_Ver'],
      dtype='object')

In [16]:
user_reviews_df.columns 

Index(['App', 'Translated_Review', 'Sentiment', 'Sentiment_Polarity',
       'Sentiment_Subjectivity'],
      dtype='object')

Already spaces are replaced by underscores

### 1.3.1 Data Cleaning part for Play store dataframe.

Firstly we will remove the duplicated rows for the play store dataframe.

In [17]:
duplicated_rows_play_store_index=play_store_df[play_store_df.duplicated()].index
print(duplicated_rows_play_store_index)

Int64Index([  229,   236,   239,   256,   261,   265,   266,   267,   268,
              269,
            ...
             8632,  8635,  8636,  8639,  8642,  8643,  8654,  8658, 10049,
            10768],
           dtype='int64', length=483)


In [18]:
# Dropping the duplicated rows as they were disturbing the data integrity and checking the shape
play_store_df.drop(duplicated_rows_play_store_index, inplace=True)
play_store_df.shape

(10358, 13)

In [19]:
play_store_df.App.isnull().sum()

0

No null values in App feature.

In [20]:
play_store_df.Category.isnull().sum()

0

No null values in Category feature.

In [21]:
# Getting the unique values of category column
play_store_df.Category.unique()

array(['ART_AND_DESIGN', 'AUTO_AND_VEHICLES', 'BEAUTY',
       'BOOKS_AND_REFERENCE', 'BUSINESS', 'COMICS', 'COMMUNICATION',
       'DATING', 'EDUCATION', 'ENTERTAINMENT', 'EVENTS', 'FINANCE',
       'FOOD_AND_DRINK', 'HEALTH_AND_FITNESS', 'HOUSE_AND_HOME',
       'LIBRARIES_AND_DEMO', 'LIFESTYLE', 'GAME', 'FAMILY', 'MEDICAL',
       'SOCIAL', 'SHOPPING', 'PHOTOGRAPHY', 'SPORTS', 'TRAVEL_AND_LOCAL',
       'TOOLS', 'PERSONALIZATION', 'PRODUCTIVITY', 'PARENTING', 'WEATHER',
       'VIDEO_PLAYERS', 'NEWS_AND_MAGAZINES', 'MAPS_AND_NAVIGATION',
       '1.9'], dtype=object)

These are all the unique categories of application that we are having in our data.

In [22]:
play_store_df.Rating.isnull().sum()

1465

We are having 1465 null values in rating column which we have to fill or drop out.

In [23]:
print("Minimum rating to an App given:",min(play_store_df.Rating))
print("Maximum rating to an App given:",max(play_store_df.Rating))

Minimum rating to an App given: 1.0
Maximum rating to an App given: 19.0


Max rating of the App can be 5.0

In [24]:
play_store_df[play_store_df.Rating>5]

App Category  ...  Current_Ver Android_Ver
10472  Life Made WI-Fi Touchscreen Photo Frame      1.9  ...   4.0 and up         NaN

[1 rows x 13 columns]

We can see this record is manipulated as it has rating more than 5, price is everyone which should be some value, installs is free etc. so we will remove this record.

In [25]:
play_store_df.drop(play_store_df[play_store_df.Rating>5].index,inplace=True)

In [26]:
print("Now maximum rating of an App is:",max(play_store_df.Rating))

Now maximum rating of an App is: 5.0


In [27]:
play_store_df.reset_index(drop=True,inplace=True)

In [28]:
play_store_df.head()

App  ...   Android_Ver
0     Photo Editor & Candy Camera & Grid & ScrapBook  ...  4.0.3 and up
1                                Coloring book moana  ...  4.0.3 and up
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ...  4.0.3 and up
3                              Sketch - Draw & Paint  ...    4.2 and up
4              Pixel Draw - Number Art Coloring Book  ...    4.4 and up

[5 rows x 13 columns]

In [29]:
play_store_df.Rating.isnull().sum()

1465

Still we are having 1465 number of null values in our Rating column.

In [30]:
versions_of_app=play_store_df[play_store_df.App.duplicated(keep=False)].groupby(by=["App"],as_index=False).apply(lambda x: x[:])
versions_of_app

App  ...         Android_Ver
0   2697                            365Scores - Live Scores  ...          4.1 and up
    4959                            365Scores - Live Scores  ...          4.1 and up
1   1468                                        8 Ball Pool  ...        4.0.3 and up
    1496                                        8 Ball Pool  ...        4.0.3 and up
    1544                                        8 Ball Pool  ...        4.0.3 and up
...                                                     ...  ...                 ...
520 2331                        textPlus: Free Text & Calls  ...          4.1 and up
521 2602  theScore: Live Sports Scores, News, Stats & Vi...  ...          4.4 and up
    2680  theScore: Live Sports Scores, News, Stats & Vi...  ...          4.4 and up
522 3226                     wetter.com - Weather and Radar  ...  Varies with device
    7820                     wetter.com - Weather and Radar  ...  Varies with device

[1221 rows x 13 columns]

In [31]:
versions_of_app[versions_of_app["Rating"].isnull()]

App        Category  ...  Current_Ver Android_Ver
48  2222          Blood Pressure         MEDICAL  ...          1.0  2.3 and up
282 23    Mcqueen Coloring pages  ART_AND_DESIGN  ...        1.0.0  4.1 and up
    1878  Mcqueen Coloring pages          FAMILY  ...        1.0.0  4.1 and up

[3 rows x 13 columns]

Checking for the different versions and filling the rating of Applications with there mean.

In [32]:
play_store_df[play_store_df["App"]=='Blood Pressure']

App Category  Rating  ...    Last_Updated Current_Ver Android_Ver
2052  Blood Pressure  MEDICAL     4.2  ...   July 24, 2018      3.27.3  4.1 and up
2222  Blood Pressure  MEDICAL     NaN  ...  March 14, 2015         1.0  2.3 and up

[2 rows x 13 columns]

Only 2 versions available so we can fill with the mean of category to which this app belongs.

In [33]:
play_store_df[play_store_df["App"]=='Mcqueen Coloring pages']

App        Category  ...  Current_Ver Android_Ver
23    Mcqueen Coloring pages  ART_AND_DESIGN  ...        1.0.0  4.1 and up
1878  Mcqueen Coloring pages          FAMILY  ...        1.0.0  4.1 and up

[2 rows x 13 columns]

Here also only 2 applications are there and they are also present in the different category so we will put catgory wise mean in both.

In rest of the records with null-values in rating column we will fill category wise mean because we don't have different versions of those applications.

In [34]:
play_store_df['Rating'].fillna(play_store_df.groupby('Category')['Rating'].transform('mean'), inplace=True)

In [35]:
# Checking the Reivew column
play_store_df['Reviews']=play_store_df['Reviews'].astype('int')

In [36]:
play_store_df['Size'].unique()[30:70]

array(['9.4M', '15M', '10M', '1.2M', '26M', '8.0M', '7.9M', '56M', '57M',
       '35M', '54M', '201k', '3.6M', '5.7M', '8.6M', '2.4M', '27M',
       '2.5M', '16M', '3.4M', '8.9M', '3.9M', '2.9M', '38M', '32M',
       '5.4M', '18M', '1.1M', '2.2M', '4.5M', '9.8M', '52M', '9.0M',
       '6.7M', '30M', '2.6M', '7.1M', '3.7M', '22M', '7.4M'], dtype=object)

As we can see we have size of the app in Mb's and Kb's we have to convert these into same unit and also remove the unwanted symbols.

In [37]:
play_store_df['Size'] = play_store_df['Size'].apply(lambda x: str(x).replace('Varies with device', 'NaN') if 'Varies with device' in str(x) else x)
play_store_df['Size'] = play_store_df['Size'].apply(lambda x: str(x).replace('M', '') if 'M' in str(x) else x)
play_store_df['Size'] = play_store_df['Size'].apply(lambda x: str(x).replace(',', '') if ',' in str(x) else x)
play_store_df['Size'] = play_store_df['Size'].apply(lambda x: float(str(x).replace('k', '')) / 1024 if 'k' in str(x) else x)
play_store_df['Size'] = play_store_df['Size'].apply(lambda x: str(x).replace('+','') if '+' in str(x) else x)

In [38]:
play_store_df['Size'].dtype

dtype('O')

In [ ]:
# Type Casting the Size feature in float64 
play_store_df.Size=play_store_df.Size.astype('float64')

In [ ]:
# Checking the data type
play_store_df['Size'].dtype

In [ ]:
play_store_df['Size'].isnull().sum()

We have 1526 null values filling them with category wise mean of the Size.

In [ ]:
play_store_df['Size'].fillna(play_store_df.groupby('Category')['Size'].transform('mean'), inplace=True)

In [ ]:
# Checking for the Installs feature
play_store_df.Installs.unique(),len(play_store_df.Installs.unique())

We can use the Installs columns category wise too and numerical wise too in our EDA section so we will not change it over here.

In [ ]:
play_store_df.Type.unique()

In [ ]:
# Type feature null values
play_store_df[play_store_df.Type.isnull()]

We have only single value so we will fill the Type in this record manually.

In [ ]:
play_store_df['Price'].unique()[5:15]

Here we can see $ sign but to make Price column categorical we have to remove it and type cast it to float64.

In [ ]:
play_store_df['Price'] = play_store_df['Price'].apply(lambda x: float(str(x).replace('$', '')) if '$' in str(x) else float(x))

In [ ]:
# Checking for the rows where Price is 0 and Type is Paid.
play_store_df[(play_store_df["Price"]==0) & (play_store_df["Type"]=="Paid")]

As we do not have any rows we will fill the single Type record with Nan value manually.

In [ ]:
play_store_df["Type"].fillna('Free',inplace=True)

In [ ]:
# Type casting the Last_updated column into Date Time Format  
from datetime import datetime
play_store_df["Last_Updated"]=play_store_df["Last_Updated"].apply(lambda x: datetime.strptime(x, "%B %d, %Y"))

In [ ]:
print(list(play_store_df.Current_Ver.unique())[10:40])

List of Current_ver unique values we can't fill the nan values in this column because Apps are having different values in versions too not in particular sequential increasing order.

In [ ]:
current_version_nan=list(play_store_df[play_store_df["Current_Ver"].isnull()]["App"])
for i in range(len(current_version_nan)):
  print(play_store_df[play_store_df['App']==current_version_nan[i]])

Only 8 values are there so we will remove as this data is less than .001% of data.

In [ ]:
# Checking null values in Android Version
play_store_df[play_store_df.Android_Ver.isnull()]

Only 2 null values.

In [ ]:
play_store_df[play_store_df.App=="Pi Dark [substratum]"]

In [ ]:
play_store_df[play_store_df.App=="[substratum] Vacuum: P"]

We don't have different versions of these application so we will fill them with most used android version in that particular Year manually.

In [ ]:
# Making a new column named Year
play_store_df["Year"]=play_store_df.Last_Updated.dt.year

In [ ]:
play_store_df[play_store_df.Year==2018]['Android_Ver'].value_counts()

Android version 4.0 and above is mostly used in Year 2018 so we will fill null values with "4.1 and up".

In [ ]:
play_store_df['Android_Ver'].fillna("4.1 and up", inplace=True)

In [ ]:
# Dropping all the records where null values are there 
play_store_df.dropna(inplace=True)

In [ ]:
play_store_df.info()

No null values are there in our Dataframe.

In [ ]:
play_store_df.describe(include="all")

### 1.3.2 Cleaning the user reviews dataframe

In [ ]:
# Cleaning user reviews dataframe
user_reviews_df.info()

In [ ]:
user_reviews_df.isnull().sum()

We will check for the null values wether they have common index or not and removing those null values.

In [ ]:
sentiment_isnull_index=user_reviews_df[user_reviews_df.Sentiment.isnull()].index
sentiment_polarity_isnull_index=user_reviews_df[user_reviews_df.Sentiment_Polarity.isnull()].index
sentiment_subjectivity_isnull_index=user_reviews_df[user_reviews_df.Sentiment_Subjectivity.isnull()].index

In [ ]:
if sentiment_isnull_index.all() == sentiment_polarity_isnull_index.all():
  print("All indexs are common in sentiment_isnull_index and sentiment_polarity_isnull_index")
if sentiment_isnull_index.all() == sentiment_subjectivity_isnull_index.all():
  print("All indexs are common in sentiment_isnull_index and sentiment_subjectivity_isnull_index")

In [ ]:
user_reviews_df.drop(sentiment_subjectivity_isnull_index, inplace=True)

In [ ]:
user_reviews_df.describe(include="all")

Dropped the common indexes.
Now our both the data frames are cleaned handling of NaN/ Null/ Missing Values is completed.

# 1.4 Extraction of features information
Here all the features with the information they are providing present in our original dataframe is there.

### Play Store dataset contains following columns and their information:
1. App: Application name
2. Category: Category to which the app belongs (34 available)
3. Rating: Rating of the app (0 - 5)
4. Reviews: Number of reviews of the app available (Discrete)
5. Size: Size of the app in MB (Discrete)
6. Installs: Number of times the app has installed (Multi categorical)
7. Type: Free or Paid
8. Price: Price of the app
9. Content Rating: Whether it is for adults, teenagers etc. based on age(6 available)
10. Generes: Genere of the app
11. Last Updated: Last Update date
12. Current Ver: Current version of the application in playstore
13. Android Ver: Android version of the application

### User Reviews dataset contains following columns and their information:
1. App: Application name (duplicates present depending upon number of reviews present).
2. Translated_Review: Translated Reviews of that application.
3. Sentiment: Sentiment of the Review whether positive, negative or neutral.
4. Sentiment_Polarity: Polarity of the sentiment -ve vs +ve (-1 to +1).
5. Sentiment_Subjectivity: Sentiment is near to application subject subjective vs objective(0 - 1).

### Adding description about the sentiment analysis
**A sentiment is defined by its semantic orientation and the intensity of each word in the sentence.** This requires a pre-defined dictionary classifying negative and positive words. Generally, a text message will be represented by bag of words. After assigning individual scores to all the words, final sentiment is calculated by some pooling operation like taking an average of all the sentiments.

TextBlob returns polarity and subjectivity of a sentence. Polarity lies between [-1,1], -1 defines a negative sentiment and 1 defines a positive sentiment. Negation words reverse the polarity. TextBlob has semantic labels that help with fine-grained analysis. 

**Subjectivity quantifies the amount of personal opinion and factual information contained in the text. The higher subjectivity means that the text contains personal opinion rather than factual information.** TextBlob has one more parameter — intensity. TextBlob calculates subjectivity by looking at the ‘intensity’. Intensity determines if a word modifies the next word. For English, adverbs are used as modifiers (‘very good’).

For example: We calculated polarity and subjectivity for “I do not like this example at all, it is too boring”. For this particular example, polarity = -1 and subjectivity is 1, which is fair.

However, for the sentence “This was a helpful example but I would prefer another one”. It returns 0.0 for both subjectivity and polarity which is not the finest answer we’d expect.

In [ ]:
print("Play store dataframe shape:",play_store_df.shape,"\nUser reviews dataframe shape:",user_reviews_df.shape)

# 1.5 Exploratory Data Analysis (EDA)

Trying to get some conclusion from data, correlation, trends which make sense and accomplishing milestones given in the problem statement by using visualization.

For each visualization we will create a new dataframe.


In [ ]:
df1=play_store_df[["Year","Category"]].groupby(["Year","Category"],as_index=False).size()

In [ ]:
fig = px.pie(df1, values='size', names='Category', hole = 0.4, title='Number of Apps present in dataframe according to Category')
fig.add_annotation(text = '2010-2018', font = dict(size=20, color='black'), showarrow = False)
fig.show()

Here we are able to see percentage of apps in each category.

In [ ]:
# Making a new column name Month
play_store_df["Month"]=play_store_df.Last_Updated.dt.month

In [ ]:
play_store_df[['Month','Year']]

In [ ]:
# Converting numerical month to textual months
import calendar
play_store_df['Month'] = play_store_df['Month'].apply(lambda x: calendar.month_name[x])

In [ ]:
play_store_df[['Month','Year']]

In [ ]:
df2=play_store_df[["Category","Year","Month"]].groupby(["Year","Month","Category"], as_index=False).size()
df2.rename(columns={"size":"NumberOfApps"},inplace=True)

In [ ]:
df2.head()

In [ ]:
month=["January", "February", "March", "April", "May", "June", 
      "July", "August", "September", "October", "November", "December"]

In [ ]:
fig = px.bar(df2[df2['Year']==2018], x = 'Month', y = "NumberOfApps", color="Category", title='2018 Monthly Category Trend', category_orders={"Month": month})
fig.show()
fig = px.bar(df2[df2['Year']==2017], x = 'Month', y = "NumberOfApps", color="Category", title='2017 Monthly Category Trend', category_orders={"Month": month})
fig.show()
fig = px.bar(df2[df2['Year']==2016], x = 'Month', y = "NumberOfApps", color="Category", title='2016 Monthly Category Trend', category_orders={"Month": month})
fig.show()
fig = px.bar(df2[df2['Year']==2015], x = 'Month', y = "NumberOfApps", color="Category", title='2015 Monthly Category Trend', category_orders={"Month": month})
fig.show()
fig = px.bar(df2[df2['Year']==2014], x = 'Month', y = "NumberOfApps", color="Category", title='2014 Monthly Category Trend', category_orders={"Month": month})
fig.show()
fig = px.bar(df2[df2['Year']==2013], x = 'Month', y = "NumberOfApps", color="Category", title='2013 Monthly Category Trend', category_orders={"Month": month})
fig.show()
fig = px.bar(df2[df2['Year']==2012], x = 'Month', y = "NumberOfApps", color="Category", title='2012 Monthly Category Trend', category_orders={"Month": month})
fig.show()
fig = px.bar(df2[df2['Year']==2011], x = 'Month', y = "NumberOfApps", color="Category", title='2011 Monthly Category Trend', category_orders={"Month": month})
fig.show()
fig = px.bar(df2[df2['Year']==2010], x = 'Month', y = "NumberOfApps", color="Category", title='2010 Monthly Category Trend', category_orders={"Month": month})
fig.show()

Here we can see monthly trend of category of applications.
We can find out in which month which type of applications are mostly released in each year(2010-2018).

In [ ]:
df3=play_store_df[["Installs","Last_Updated","Year","Month","Category"]].groupby(["Year","Month","Installs","Category","Last_Updated"], as_index=False).size()
df3

In [ ]:
# Here we are converting our Installs into numerical column
df3['Installs'] = df3['Installs'].apply(lambda x: str(x).replace('+','') if '+' in str(x) else x)
df3['Installs'] = df3['Installs'].apply(lambda x: str(x).replace(',', '') if ',' in str(x) else x)
df3['Installs']=df3['Installs'].astype('int')

In [ ]:
fig = px.bar(df3[df3['Year']==2018], x = "Month", y = "Installs", color="Category", title='2018 Monthly installation Trend', category_orders={"Month": month})
fig.show()
fig = px.bar(df3[df3['Year']==2017], x = 'Month', y = "Installs", color="Category", title='2017 Monthly installation Trend', category_orders={"Month": month})
fig.show()
fig = px.bar(df3[df3['Year']==2016], x = 'Month', y = "Installs", color="Category", title='2016 Monthly installation Trend', category_orders={"Month": month})
fig.show()
fig = px.bar(df3[df3['Year']==2015], x = 'Month', y = "Installs", color="Category", title='2015 Monthly installation Trend', category_orders={"Month": month})
fig.show()
fig = px.bar(df3[df3['Year']==2014], x = 'Month', y = "Installs", color="Category", title='2014 Monthly installation Trend', category_orders={"Month": month})
fig.show()
fig = px.bar(df3[df3['Year']==2013], x = 'Month', y = "Installs", color="Category", title='2013 Monthly installation Trend', category_orders={"Month": month})
fig.show()
fig = px.bar(df3[df3['Year']==2012], x = 'Month', y = "Installs", color="Category", title='2012 Monthly installation Trend', category_orders={"Month": month})
fig.show()
fig = px.bar(df3[df3['Year']==2011], x = 'Month', y = "Installs", color="Category", title='2011 Monthly installation Trend', category_orders={"Month": month})
fig.show()
fig = px.bar(df3[df3['Year']==2010], x = 'Month', y = "Installs", color="Category", title='2010 Monthly installation Trend', category_orders={"Month": month})
fig.show()

Here we are getting monthly trend of each category installed in each year which can provide us the installation details according festivals, holiday or political shift etc.

To get it in more better way we can compare month wise for 2018 and 2017 in the next graphs.

In [ ]:
df4=df3[["Year","Month","Installs","Category"]].groupby(["Year","Month","Category"],as_index=False).sum()

In [ ]:
fig = px.bar(df4[df4['Year']==2018], x = "Category", y = "Installs", animation_frame="Month",text="Category", animation_group="Category", title='2018 Monthly installation Trend', category_orders={"Month": month})
fig.update_traces(textposition='outside')
fig.update_xaxes(visible=False)
fig.show()

In [ ]:
fig = px.bar(df4[df4['Year']==2017], x = "Category", y = "Installs", animation_frame="Month", animation_group="Category", text="Category", title='2017 Monthly installation Trend', category_orders={"Month": month})
fig.update_traces(textposition='outside')
fig.update_xaxes(visible=False)
fig.show()

Monthly trend of installations of different categories.

In [ ]:
df5=play_store_df[["App","Year","Category","Type","Installs","Price"]].groupby(["Year","Category","Installs","Type","Price","App"], as_index=False).size()


In [ ]:
fig = px.scatter(df5, x = "Year", y = "Price", color="Category", symbol="Category", title='Price of different Apps of different Categories from 2010 to 2018')
fig.show()

Here we have the different prices of the Apps of different categories yearly. We can see most of the apps have prices below 50$.

In [ ]:
df6=play_store_df[["Year","Category","Rating"]].groupby(["Year","Category"],as_index=False).mean()

In [ ]:
fig = px.bar(df6, x = "Category", y = "Rating",animation_frame="Year", animation_group="Category",text="Category",title="Category wise mean Rating from 2010 to 2018")
fig.update_xaxes(visible=False)
fig.show()

As we have Rating of all the Categories in 2018 is above 4.0 but in 2017 we have some of the Apps below 4.0 so we can check the popularity of different categories of Apps in different years.

In [ ]:
df7=play_store_df[["Year","Android_Ver"]].groupby(["Year","Android_Ver"],as_index=False).size()

In [ ]:
fig = px.bar(df7, x = "Android_Ver", y = "size",animation_frame="Year", animation_group="Android_Ver",text="Android_Ver",title='Popular android version in different year')
fig.update_xaxes(visible=False)
fig.show()